src/ray/raylet/scheduling/policy/scorer.h

```diff
@@ -0,0 +1,56 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+#include <optional>
+
+#include "ray/raylet/scheduling/cluster_resource_data.h"
+
+namespace ray {
+namespace raylet_scheduling_policy {
+
+/// NodeScorer is a scorer to make a grade to the node, which is used for scheduling
+/// decision.
+class NodeScorer {
```

Comment:
do we have unit test for this?

Reply:
> do we have unit test for this?

These codes comes from `gcs_resource_scheduler`, this PR aims to remove `gcs_resource_scheduler`, so I just move the logic inside `gcs_resource_scheduler` to `scorer.h/.cc` and `bundle_scheduling_policy.h/.cc`.